unify_idents/engine_parsers/ident/omssa_2_1_9_parser.py

```diff
@@ -0,0 +1,262 @@
+"""Engine parser."""
+import xml.etree.ElementTree as ETree
+
+import numpy as np
+import pandas as pd
+
+from unify_idents.engine_parsers.base_parser import __IdentBaseParser
+
+
+class OmssaParser(__IdentBaseParser):
+    """File parser for OMSSA."""
+
+    def __init__(self, *args, **kwargs):
+        """Initialize parser.
+
+        Reads in data file and provides mappings.
+        """
+        super().__init__(*args, **kwargs)
+        self.style = "omssa_style_1"
+
+        self.df = pd.read_csv(self.input_file)
+
+        self.mapping_dict = {
+            v: k
+            for k, v in self.param_mapper.get_default_params(style=self.style)[
+                "header_translations"
+            ]["translated_value"].items()
+        }
+        self.df.rename(columns=self.mapping_dict, inplace=True)
+        self.df.columns = self.df.columns.str.lstrip(" ")
+        self.df.drop(
+            columns=[
+                c
+                for c in self.df.columns
+                if c
+                not in set(self.mapping_dict.values()) | set(self.reference_dict.keys())
+            ],
+            inplace=True,
+            errors="ignore",
+        )
+        self.reference_dict.update({k: None for k in self.mapping_dict.values()})
+
+    @classmethod
+    def check_parser_compatibility(cls, file):
+        """Assert compatibility between file and parser.
+
+        Args:
+            file (str): path to input file
+
+        Returns:
+            bool: True if parser and file are compatible
+
+        """
+        is_csv = file.as_posix().endswith(".csv")
+        with open(file.as_posix()) as f:
+            head = "".join([next(f) for _ in range(1)])
+        head = set(head.rstrip("\n").split(","))
+        ref_columns = {
+            "Spectrum number",
+            " Filename/id",
```

Comment:
What is there to hate about OMSSA spacing in the column headers? I'd say, it adds some spice to it ! :)